singleheader/simdjson.cpp

```diff
@@ -697,16 +697,10 @@ really_inline bool add_overflow(uint64_t value1, uint64_t value2, uint64_t *resu
 #endif
 }
 
-#ifdef _MSC_VER
-#pragma intrinsic(_umul128) // todo: this might fail under visual studio for ARM
-#endif
-
 really_inline bool mul_overflow(uint64_t value1, uint64_t value2, uint64_t *result) {
 #ifdef _MSC_VER
-  // todo: this might fail under visual studio for ARM
-  uint64_t high;
-  *result = _umul128(value1, value2, &high);
-  return high;
+  *result = value1 * value2;
+  return !!__umulh(value1, value2);
```

Comment:
This is probably fine on ARM 64, but it is suboptimal on x64. Here is what it compiles to:
```asm
        mov     rax, rcx
        imul    rax, rdx
        mov     QWORD PTR [r8], rax
        mov     rax, rdx
        mul     rcx
        test    rdx, rdx
        setne   al
```
Notice the two multiplications?

If you try, instead the following...
```C++
  uint64_t high; 
  *result = _umul128(value1, value2, &high);		
   return high;
```
You get the much better assembly on x64...

```asm
        mov     rax, rdx
        mul     rcx
        test    rdx, rdx
        mov     QWORD PTR [r8], rax
        setne   al
```

Reply:
That block is completely in simdjson:: arm64 namespace. Arm-only code for _MSC_VER